packages/clay-shared/src/useTransitionHeight.ts

```diff
@@ -40,16 +40,42 @@ export function useTransitionHeight(
 ) {
 	const [transitioning, setTransitioning] = React.useState<boolean>(false);
 
+	let transitionTimer: any;
```

Comment:
This is very unsafe, as we can easily lose its value in a rendering, to maintain the ideal would be to use `useRef` here.

Reply:
Not sure how to use `useRef` for this case, this variable is to hold the `setTimeout` function so we can clear it in `handleTransitionEnd`. Can you give me an example?

Also I think it's better to manually fire a React `transitionEnd` event instead of reimplementing `handleTransitionEnd` in the `setTimeout`, but I couldn't find any documentation on it.

Reply:
> Not sure how to use useRef for this case, this variable is to hold the setTimeout function so we can clear it in handleTransitionEnd. Can you give me an example?

Sure, see this example:

```js
const transitionTimerRef = useRef<NodeJS.Timeout | null>(null);

// Write
transitionTimerRef.current = setTimeout

// Read
clearTimeout(transitionTimerRef.current);
```

> Also I think it's better to manually fire a React transitionEnd event instead of reimplementing handleTransitionEnd in the setTimeout, but I couldn't find any documentation on it.

To do this we would have to manually trigger an event on the node, but we would still have to have a timeout anyway. Starting it immediately could break the animation.

Reply:
One note is sometimes TS complains about the type of `setTimeout`, which is why I think `NodeJS.Timeout` is used as the type. But just be aware it may complain